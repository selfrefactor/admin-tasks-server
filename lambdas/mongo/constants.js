const { resolve } = require('path')
exports.DATA_LOCATION = resolve(__dirname, '../../../data')
