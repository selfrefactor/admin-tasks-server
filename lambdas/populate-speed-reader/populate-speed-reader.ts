import * as extractZip from 'extract-zip'
import {parse, resolve} from 'path'
import {scanFolder} from 'helpers-fn'
import { move } from 'fs-extra'
import {
  endsWith,
  init,
  mapAsync,
  match,
  ok,
  piped,
  remove,
  replace,
  test,
  toLower,
} from 'rambdax'

const extractDir = `${__dirname}/extracted`

const destinationDir = resolve(
  __dirname,
  '../../../data/books'
)

export function generateFileName(filePath){
  const matched = match(
    /assets\/[a-zA-Z_\.]+-[a-zA-Z_\.]+/
  )(filePath)

  if(matched.length === 0) throw new Error('either file path or regex is wrong')

  const result = piped<string>(
    matched[0],
    remove(
      ['assets/', 'b.txt.zip']
    ),
    replace('_-_','.'),
    replace(/_/g,'.'),
    replace(/\.\./g,'.'),
    x => x.endsWith('.') ? init(x) : x,
    toLower
  )

  const okResult = test(
    /^[a-z\.]+$/
  )(result)

  ok(okResult)(true)

  return result    
}

async function processSingleZip(zipFilePath){
  await extractZip(zipFilePath, { dir: extractDir })
  const fileName = generateFileName(zipFilePath)
  const {name: originFileName} = parse(zipFilePath)
  
  const fullPath = `${extractDir}/${originFileName}`

  await move(
    fullPath,
    `${destinationDir}/${fileName}.txt`,
    {overwrite: true}
  )

  return {fileName, fullPath}
}

export async function populateSpeedReader() {
  try {
    const assetsDir = `${__dirname}/assets`
    const allZips = await scanFolder({
      folder: assetsDir,
      filterFn: endsWith('.zip')
    }) 

    await mapAsync(processSingleZip)(allZips)
  } catch (error) {
    console.log(error, 'populate.speed.reader')
  }
}
 