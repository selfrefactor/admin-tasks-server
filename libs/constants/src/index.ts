import {log} from 'helpers-fn'
export const REPO_LOCATION = `${process.env.HOME}/repos`
export const DATA_LOCATION = `${REPO_LOCATION}/data`
export const DEFAULT_PORT = 4042
export const DEV_PORT = 8087
const WORK_NGROK_PORT = 3000
export const NGROK_PORT = process.env.WORK === 'ON' ? WORK_NGROK_PORT : DEFAULT_PORT
export const IS_WALLABY = process.env.WALLABY_PRODUCTION !== undefined

export const getWordProfileConnectUrl = () => {
  if (!process.env.WORD_PROFILE_MONGO_CONNECT_URL) {
    log('process will exit', 'error')
    process.exit()
  }

  return process.env.WORD_PROFILE_MONGO_CONNECT_URL
}
