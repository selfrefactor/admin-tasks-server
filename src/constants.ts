export const REPOS_DIR = `${ process.env.HOME }/repos`