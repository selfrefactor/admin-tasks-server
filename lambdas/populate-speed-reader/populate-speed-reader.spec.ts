import {populateSpeedReader} from './populate-speed-reader'
import { IS_JEST } from 'lib/constants'

test('happy', async() => {
  if(!IS_JEST) return
  await populateSpeedReader()
})
