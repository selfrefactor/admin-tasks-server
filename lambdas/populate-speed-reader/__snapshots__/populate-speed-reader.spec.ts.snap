// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate file name 1`] = `
Array [
  "ajn.rand.romanticheski.manifest",
  "aleksandyr.girginov.smjah.i.mydrost.prez.vekovete",
  "anton.pavlovich.chehov.damata.s.kuchentseto",
  "bob.fenstyr.bezkrajnostta.na.choveshkata.prostotija",
  "bob.fenstyr.oshte.prostotii",
  "bogomil.gerasimov.ot.abyrdijn.s.usmivka.shotlandski.humor",
  "fridrih.nitsshe.ecce.homo.kak.se.stava.takyv.kakyvto.si",
  "fxodor.dostoevski.beli.noshti.santimentalen.roman",
  "henri.milyr.nexus",
  "lao.dzy.dao.dy.dzin",
  "lao.dzy.pytjat",
  "lev.tolstoj.otets.sergij",
  "lev.tolstoj.vojna.i.mir.treti.i.chetvyrti.tom",
  "lev.tolstoj.vyzkresenie",
  "nikolaj.gogol.peterburgski.povesti",
  "nikolaj.gogol.povesti",
  "nikolaj.gogol.vecheri.v.seltseto.kraj.dikanka",
  "p.g.udhaus.bankeri.po.nevolja",
  "p.g.udhaus.bil.zavoevatelja",
  "p.g.udhaus.chicho.fred.prez.proletta",
  "p.g.udhaus.dzhijvs.zapretva.rykavi",
  "p.g.udhaus.dzhim.tajfuna",
  "p.g.udhaus.gafovete.na.archi",
  "p.g.udhaus.galahad.v.blandings",
  "p.g.udhaus.goljamoto.nakisvane",
  "p.g.udhaus.ikonom.za.edin.den",
  "p.g.udhaus.klub.anonimni.ergeni",
  "p.g.udhaus.kukovo.ljato.v.blandings",
  "p.g.udhaus.pelikan.v.blandings",
  "p.g.udhaus.skrito.pokrito",
  "p.g.udhaus.zakonyt.na.ustyr",
  "temida.se.sheguva.anekdoti.i.neverojatni.istorii.ot.praktikata.na.juristite",
  "volter.filosofski.noveli",
]
`;
